From eff7a8bafe3f9b1113c4fa3824480b399c264c45 Mon Sep 17 00:00:00 2001
From: Athos Ribeiro <athoscr@fedoraproject.org>
Date: Sat, 4 Jul 2020 17:24:05 +0200
Subject: [PATCH] Remove calls to deprecated plistlib function

plistlib's readPlist has been deprecated since python 3.4 and was
removed in python 3.9. We substitute it for direct calls to the plist
load function.

Signed-off-by: Athos Ribeiro <athoscr@fedoraproject.org>
---
 firehose/parsers/clanganalyzer.py | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/firehose/parsers/clanganalyzer.py b/firehose/parsers/clanganalyzer.py
index a8f98b0..28cdd71 100644
--- a/firehose/parsers/clanganalyzer.py
+++ b/firehose/parsers/clanganalyzer.py
@@ -40,12 +40,13 @@ def parse_scandir(resultdir, analyzerversion=None, sut=None):
     for filename in glob.glob(os.path.join(resultdir, 'report-*.plist')):
         yield parse_plist(filename, analyzerversion, sut)
 
-def parse_plist(pathOrFile, analyzerversion=None, sut=None, file_=None, stats=None):
+def parse_plist(file_path, analyzerversion=None, sut=None, file_=None, stats=None):
     """
     Given a .plist file emitted by clang-static-analyzer (e.g. via
     scan-build), parse it and return an Analysis instance
     """
-    plist = plistlib.readPlist(pathOrFile)
+    with open(file_path, 'rb') as f:
+        plist = plistlib.load(f)
     # We now have the .plist file as a hierarchy of dicts, lists, etc
 
     # Handy debug dump:
-- 
2.26.2

